lib/handleRequest.js

```diff
@@ -77,7 +77,7 @@ function handler (request, reply) {
 function preValidationCallback (err, request, reply) {
   if (reply.sent === true ||
     reply.raw.writableEnded === true ||
-    (reply.raw.writable === false && reply.raw.finished === true)) return
```

Comment:
`reply.raw.finished` never gets covered, this makes me think that checking `writable` should be enough.

This is something I added in https://github.com/fastify/fastify/pull/2233